restricted_QAQC/global.R

```diff
@@ -210,33 +219,51 @@ con = dbConnect(y,
                 dbname = dbname)
 tables = dbListTables(con)
 
-# # Code for one-time use: to load data into mysql
-# dataCurrent <- read.csv("data/current_clean20181202.csv", stringsAsFactors = FALSE, na.strings=c(""," ", "NA"))
+# Code for one-time use: to load data into mysql
+# dataCurrent <- read.csv("data/current_clean.csv", stringsAsFactors = FALSE, na.strings=c(""," ", "NA"))
 #  dataCurrent$date <- as.Date(dataCurrent$date, "%m/%d/%y")
 #  dataCurrent <- standardizeClasses(dataCurrent)
 #  dbWriteTable(con, "current", dataCurrent, append = TRUE, row.names = FALSE)
-#
+# 
 # dataHistorical<- read.csv("data/historical.csv", stringsAsFactors = FALSE, na.strings=c(""," ", "NA"))
 #  dataHistorical$date <- as.POSIXct(dataHistorical$date, "%Y-%m-%d")
 #  dataHistorical <- standardizeClasses(dataHistorical)
 #  dbWriteTable(con, "historical", dataHistorical, append = TRUE, row.names = FALSE)
 
-# Get data from mysql
-dataCurrent <- dbReadTable(con, "current") %>%
+calculate_SWDD <- function(data) {
```

Comment:
This function is good, but I forgot to define degree days as they pertain to HBEF. This page explains that for the HBEF ecosystem study, a "degree day" is the number of degrees over 4 Celsius reached by the maximum temperature that day. We're doing the same thing for water. And we can use the same threshold of 4 degrees. I haven't tested this, but here's GPT's solution:

calculate_SWDD <- function(data, base_temp_C = 4) {
    data %>%
        mutate(date = as.Date(date)) %>%  
        arrange(site, date) %>%
        group_by(site, year = lubridate::year(date)) %>%  
        mutate(
            temp_diff = pmax(temp - base_temp, 0, na.rm = TRUE), # Only positive differences for heating degree days
            dd = cumsum(temp_diff)
        ) %>%
        ungroup() 
}